SetDevCommandPrompt.cmd

```diff
@@ -1,12 +1,12 @@
 @echo off
 
-:: prefer building with Dev15
-
+:: Prefer building with Dev15 and try the simple route first (we may be running from a DevCmdPrompt already)
 set CommonToolsDir=%VS150COMNTOOLS%
 
-:: VS150COMNTOOLS is not set globally any more, so fall back to default preview location before trying Dev14
-if not exist "%CommonToolsDir%" set CommonToolsDir=%ProgramFiles(x86)%\\Microsoft Visual Studio\\VS15Preview\\Common7\\Tools\\
+:: VS150COMNTOOLS is not set globally any more, so fall back to using the managed query api before trying Dev14
+if not exist "%CommonToolsDir%" for /f "usebackq delims=" %%v in (`powershell -noprofile -executionPolicy Bypass -file "%RoslynRoot%build\scripts\locate-vs.ps1"`) do set "CommonToolsDir=%%v"
 if not exist "%CommonToolsDir%" set CommonToolsDir=%VS140COMNTOOLS%
 if not exist "%CommonToolsDir%" exit /b 1
 
 call "%CommonToolsDir%\VsDevCmd.bat"
```

Comment:
So we need to file a bug on VS command prompt team, but add pushd %~dp0 popd in the mean time.

---
In reply to: [86263703](https://github.com/dotnet/roslyn/pull/14914#discussion_r86263703) [](ancestors = 86263703)